crates/tor-dirmgr/src/authority.rs

```diff
@@ -55,6 +58,36 @@ impl Authority {
     }
 }
 
+impl FromStr for Authority {
+    type Err = Error;
+
+    /// Parse Authority from a string.
+    fn from_str(authority_raw: &str) -> Result<Self, Self::Err> {
+        // name, v3ident_raw
+        let authority: Vec<&str> = authority_raw.split_whitespace().collect();
+
+        if authority.len() != 2 {
+            bail!(format!(
+                "Invalid format for authority, expected 2 elements, found {}.",
+                authority.len()
+            ));
+        }
+
+        let name = authority[0];
+        let v3ident_raw = authority[1];
```

Comment:
you can use the `if let` notation to deconstruct values; I like it but it's a matter of taste

```suggestion
        if let [name, v3ident_raw] = authority[..] {
            ...
        } else {
            bail!(format!(
                "Invalid format for authority, expected 2 elements, found {}.",
                authority.len()
            ))
        }
```